engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -1232,50 +1245,55 @@ namespace dmGameSystem
             // 2 *-*-----*-*
             //   | |  w  | |
             // 3 *-*-----*-*
-            float us[4], vs[4], xs[4], ys[4];
 
             // v are '1-v'
             xs[0] = ys[0] = 0;
             xs[3] = ys[3] = 1;
-
-            // disable slice9 computation below a certain dimension
-            // (avoid div by zero)
-            const float s9_min_dim = 0.001f;
-
-            const float su = 1.0f / org_width;
-            const float sv = 1.0f / org_height;
-            const float sx = size.getX() > s9_min_dim ? 1.0f / size.getX() : 0;
-            const float sy = size.getY() > s9_min_dim ? 1.0f / size.getY() : 0;
-
-            static const uint32_t uvIndex[2][4] = {{0,1,2,3}, {3,2,1,0}};
-            bool uv_rotated = tc[0] != tc[2] && tc[3] != tc[5];
-            bool flip_u, flip_v;
-            GetNodeFlipbookAnimUVFlip(scene, node, flip_u, flip_v);
-            if(uv_rotated)
+            bool uv_rotated;
+            if(tc)
```

Comment:
SHouldn't need to check for tc again, since you can do that earlier, and continue if it was 0.

Reply:
This is part of the bug fix though, we do need to check this here in case we have set the texture **and** 9-slice params from the script. So that this box node with custom texture will be 9-sliced.

Reply:
What I mean is, if you check for "if (!tc)" earlier, and early out with the simple case with 6 vertices, and no texture, you don't have to do "if (tc)" here. And you don't have to do the simple case twice (code duplication)

Reply:
Answered: https://github.com/defold/defold/pull/4921#issuecomment-644750126